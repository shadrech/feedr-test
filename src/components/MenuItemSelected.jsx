import React from 'react';

export const MenuItemSelected = ({ menuItem, unselectItem, dragRef, draggableProps, remove }) => {
  return (
    <li ref={dragRef} {...draggableProps} className="item selected">
      <h2>{menuItem.name}</h2>
      <p>
        {menuItem.dietaries.map(diet => <span key={diet} className="dietary">{diet}</span>)}
      </p>
      <button className="remove-item" onClick={remove}>x</button>
    </li>
  )
}
