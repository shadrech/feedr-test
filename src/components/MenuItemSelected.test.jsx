import React from 'react'
import { shallow, configure } from 'enzyme';
import { MenuItemSelected } from './MenuItemSelected';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<MenuItemSelected />', () => {

  it('renders a menu item correctly', () => {
    const item = {
      id: 1,
      name: 'Test item',
      dietaries: ['gf', 'df'],
    }
    const removeFn = jest.fn()
    const component = shallow(<MenuItemSelected menuItem={item} draggableProps={{}} remove={removeFn} />)
    expect(component.html()).toMatchSnapshot()
  })

  it('renders a menu item correctly', () => {
    const item = {
      id: 1,
      name: 'Test item',
      dietaries: ['gf', 'df'],
    }
    const removeFn = jest.fn()
    const component = shallow(<MenuItemSelected menuItem={item} draggableProps={{}} remove={removeFn} />)
    component.find('.remove-item').simulate('click')
    expect(removeFn).toHaveBeenCalledTimes(1)
  })

})
