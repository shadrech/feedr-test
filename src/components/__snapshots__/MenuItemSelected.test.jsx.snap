// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuItemSelected /> renders a menu item correctly 1`] = `"<li class=\\"item selected\\"><h2>Test item</h2><p><span class=\\"dietary\\">gf</span><span class=\\"dietary\\">df</span></p><button class=\\"remove-item\\">x</button></li>"`;
